rj_gameplay/stp/situation/__init__.py

```diff
@@ -29,10 +29,16 @@ def analyze_situation(
 
 
 class IPlaySelector(ABC):
-    """Interface for play selector."""
+    """Abstract class for play selector."""
+    def __init__(self):
+        self.analyzer: IAnalyzer = None
+        self.curr_situation: ISituation = None
+        self.curr_play: IPlay = None
 
     @abstractmethod
-    def select(self, world_state: rc.WorldState) -> Tuple[ISituation, stp.play.IPlay]:
+    def select(
+        self, world_state: rc.WorldState
+    ) -> Tuple[Optional[ISituation], IPlay]:
```

Comment:
```
Error: rj_gameplay/rj_gameplay/gameplay_node.py:118: error: "IPlaySelector" has no attribute "curr_situation"
```

This one is a little complex. I made IPlaySelector an abstract class instead of an interface by defining the init, which gives all subclasses an analyzer, curr_situation, and curr_play variable. I also changed the select method earlier so I match that change here. 

The reasoning behind the abstract class is that we have TestPlaySelector and BasicPlaySelector, which both extend IPlaySelector and need these three vars.